test/integration/targets/iam_user/tasks/main.yml

```
@@ -0,0 +1,228 @@
+---
+- name: set up aws connection info
+  module_defaults:
+    group/aws:
+      aws_access_key: "{{ aws_access_key }}"
+      aws_secret_key: "{{ aws_secret_key }}"
+      security_token: "{{ security_token | default(omit) }}"
+      region: "{{ aws_region }}"
+  block:
+
+  - name: ensure improper usage of parameters fails gracefully
+    iam_user_info:
+      path: '{{ test_path }}'
+      group: '{{ test_group }}'
+    ignore_errors: yes
+    register: iam_user_info_path_group
+  - assert:
+      that:
+        - iam_user_info_path_group is failed
+        - 'iam_user_info_path_group.msg == "parameters are mutually exclusive: group|path"'
+
+  - name: ensure exception handling fails as expected
+    iam_user_info:
+      region: 'bogus'
+      path: ''
+    ignore_errors: yes
+    register: iam_user_info
+  - assert:
+      that:
+        - iam_user_info is failed
+        - '"user" in iam_user_info.msg'
+
+  - name: ensure exception handling fails as expected with group
+    iam_user_info:
+      region: 'bogus'
+      group: '{{ test_group }}'
+    ignore_errors: yes
+    register: iam_user_info
+  - assert:
+      that:
+        - iam_user_info is failed
+        - '"group" in iam_user_info.msg'
+
+  - name: ensure exception handling fails as expected with default path
+    iam_user_info:
+      region: 'bogus'
+    ignore_errors: yes
+    register: iam_user_info
+  - assert:
+      that:
+        - iam_user_info is failed
+        - '"path" in iam_user_info.msg'
+
+  - name: ensure ansible user exists
+    iam_user:
+      name: '{{ test_user }}'
+      state: present
+    register: iam_user
+
+  - name: ensure the info used to validate other tests is valid
+    set_fact:
+        test_iam_user: '{{ iam_user.iam_user.user }}'
+  - assert:
+      that:
```

Since these are also tests for iam_user we should at least run a basic test here:

```suggestion
      that:
        - 'iam_user is changed'
```

Since your PR isn't for iam_user I don't think it's worth holding it up on thorough testing of iam_user, especially with an iam_user boto3 PR in flight.